controller/src/main/java/io/pravega/controller/server/eventProcessor/requesthandlers/UpdateStreamTask.java

```diff
@@ -84,11 +97,55 @@ public UpdateStreamTask(final StreamMetadataTasks streamMetadataTasks,
+                                                             final VersionedMetadata<EpochTransitionRecord> etr,
+                                                             final OperationContext context,
+                                                             final ScheduledExecutorService executor,
+                                                             final long requestId) {
+        return streamMetadataStore.getActiveEpoch(scope, stream, context, true, executor)
+                              .thenCompose(activeEpoch -> {
+                                  ScalingPolicy scalingPolicy = config.getStreamConfiguration().getScalingPolicy();
+                                  int minNumSegments = scalingPolicy.getMinNumSegments();
+
+                                  if ((scalingPolicy.getScaleType() == FIXED_NUM_SEGMENTS &&
+                                          activeEpoch.getSegments().size() != minNumSegments) ||
+                                          activeEpoch.getSegments().size() < minNumSegments) {
+                                      return processScale(scope, stream, minNumSegments, etr, activeEpoch, context, requestId);
+                                  } else {
+                                      return CompletableFuture.completedFuture(null);
+                                  }
+                              });
+    }
+
+    private CompletableFuture<Void> processScale(String scope, String stream, int numSegments,
+                                                 VersionedMetadata<EpochTransitionRecord> etr, EpochRecord activeEpoch,
+                                                 OperationContext context, long requestId) {
+        // First reset the existing epoch transition (any submitted but unattempted scale 
+        // request is discarded). Then submit a new scale to seal all segments in active epoch and create new segments
+        // with count = numSegments and range equally divided among them. 
+        final double keyRangeChunk = 1.0 / numSegments;
+        
+        List<Map.Entry<Double, Double>> newRange = IntStream.range(0, numSegments).boxed()
+                                                            .map(x -> new AbstractMap.SimpleEntry<>(x * keyRangeChunk, (x + 1) * keyRangeChunk))
+                                                            .collect(Collectors.toList());
+        log.debug("{} Scaling stream to update minimum number of segments to {}", requestId, numSegments);
+        return streamMetadataStore.resetEpochTransition(scope, stream, etr, context, executor)
+              .thenCompose(reset -> streamMetadataStore.submitScale(scope, stream, new ArrayList<>(activeEpoch.getSegmentIds()), newRange,
+                      System.currentTimeMillis(), reset, context, executor))
+              .thenCompose(updated -> streamMetadataTasks.processScale(scope, stream, updated, context, requestId, streamMetadataStore)
```

Comment:
Is this being treated as a manual scale here? Also, why is `io.pravega.controller.store.stream.StreamMetadataStore#startScale` not required here?

Reply:
startScale simply does some state sanity check for state being SCALING. And migrates epoch transition record if required. 
Here we are actually resetting the epoch transition explicitly to EMPTY whereby discarding any outstanding scale requests because we are going to replace all the segments with new set of segments matching the new policy so any outstanding scale requests are irrelevant and discardable.